### ISSUE TITLE
Listing of not-yet-decompiled contracts seems to stall with 504/502 occasionally

### ISSUE THREAD

**Issue by kolinko** [poanetwork/blockscout#1951]
On the current staging server, going to:

/api?module=contract&action=listcontracts&offset=100&page=0&filter=4

Seems to produce 502/504 right now, and for the past 2 hours. We are retrying connection once every 10s, and sometimes it goes through, but most often it does not.

By the way:
- offset - this is really 'limit', right?

- is there an upper reasonable limit of returned entries? we're grabbing the batches of 10k contracts, but would not want to stress your database

**Comment by zachdaniel** [comment]
Offset is limit, correct. I don’t know why it is that way, but that’s the way it is in the old endpoints, so the new rpc endpoint mimics it for consistency. I think it enforces a maximum page size of 10k. @vbaranov did we ship the change to not render contract source code and decompiled source code to stg? If not, then this issue will
Most likely be solved by that.

**Comment by vbaranov** [comment]
>did we ship the change to not render contract source code and decompiled source code to stg?

@zachdaniel we did

**Comment by zachdaniel** [comment]
Okay, looks like there is a performance issue when filtering for `4`, I will look into it.

**Comment by kolinko** [comment]
Offset = limit, understood :)

Performance issue - yes. Right now, I checked, the system didn't manage to get anything but 504 for a few hours, so the upload stalled after 10k contracts. I checked with lower offset (100 compared to the usual 2k), and it's not working either:

/api?module=contract&action=listcontracts&offset=100&page=0&filter=4

The majority of sensible contract decompilations are uploaded already anyway (there are 2-4M ctxs not uploaded yet I guess, but 99% of those remaining are the ones with empty bytecodes, etc). Still - it would be good to have a fix for this api.
